File: frontend/ZuhooCMS/src/app/modules/hrm/services/recruitment-kpi.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { RecruitmentKpiSummary } from '../models/hrm.model';

@Injectable({ providedIn: 'root' })
export class RecruitmentKpiService {
  private readonly endpoint = '/recruitment/kpis';

  constructor(private api: ApiService) {}

  // from/to are 'yyyy-MM-dd' strings from a native date input - both optional, all-time when omitted.
  // minScore (0-100) only narrows the Top Evaluated Candidates list, not the rest of the report.
  getSummary(from?: string, to?: string, minScore?: number): Observable<RecruitmentKpiSummary> {
    const params: Record<string, string | number> = {};
    if (from) params['from'] = from;
    if (to) params['to'] = to;
    if (minScore != null) params['minScore'] = minScore;
    return this.api.get<RecruitmentKpiSummary>(this.endpoint, params);
  }
}

File: backend/ZuhooCMS/src/main/java/com/zuhoocms/modules/hrm/recruitment/kpi/RecruitmentKpiService.java
package com.zuhoocms.modules.hrm.recruitment.kpi;

import java.time.LocalDate;

public interface RecruitmentKpiService {

    /**
     * Date bounds are optional/inclusive and filter on JobApplication's applied
     * date; null/null means all-time. minScore is optional and only narrows the
     * Top Evaluated Candidates list, not the rest of the report.
     */
    RecruitmentKpiResponse getSummary(LocalDate from, LocalDate to, Double minScore);
}
